core/autoscaling.go

```diff
@@ -62,6 +63,59 @@ func (a *autoScalingGroup) loadLaunchConfiguration() (*launchConfiguration, erro
 	return a.launchConfiguration, nil
 }
 
+func (a *autoScalingGroup) loadLaunchTemplate() (*launchTemplate, error) {
+	//already done
+	if a.launchTemplate != nil {
+		return a.launchTemplate, nil
+	}
+
+	lt := a.LaunchTemplate
+
+	if lt == nil {
+		return nil, errors.New("missing launch template")
+	}
+
+	ltID := lt.LaunchTemplateId
+	ltVer := lt.Version
+
+	if ltID == nil || ltVer == nil {
+		return nil, errors.New("missing launch template")
+	}
+
+	svc := a.region.services.ec2
+
+	params := &ec2.DescribeLaunchTemplateVersionsInput{
+		LaunchTemplateId: ltID,
+		Versions:         []*string{ltVer},
+	}
+
+	resp, err := svc.DescribeLaunchTemplateVersions(params)
+
+	if err != nil {
+		logger.Println(err.Error())
+		return nil, err
+	}
+
+	ltv := resp.LaunchTemplateVersions[0]
```

Comment:
I would have a check above in case the length of resp.LaunchTemplateVersions is zero

Reply:
addressed in https://github.com/AutoSpotting/AutoSpotting/pull/446/commits/c1153e58d76ad5d84dd1e5a30a0ddfaa823f26c0